app/models/transaction.rb

```diff
@@ -0,0 +1,8 @@
+class Transaction < ApplicationRecord
+  has_many :category_transactions, foreign_key: :transaction_id, dependent: :destroy
+  has_many :categories, through: :category_transactions
+  belongs_to :user
+
+  validates :name, presence: true
+  # validates :amount, numericality: { only_integer: true  }
```

Comment:
Please remove commented code from here as it can make the whole project unclean.